### PR TITLE
Add typescript

### PR DESCRIPTION
**Pull request by akattow**
Hey Joel! Here's an initial install of Typescript. I've converted the Universe as an initial example - I think it could benefit from strong typing & make it easier to make updates/changes in the future. There's definitely still improvements to be made (see: way too many `any` types for starters!) but this is functional without any major changes to the existing code.

Let me know what you think! I'd love to make some more updates if you dig this (the Universe is fun!)

### CODE REVIEW

**Comment by joelhans** [comment]
Hey @akattow! Thanks so much for this, and for your patience as I work my way through being burned out from development work after we launched. So many things to fix! But I'm making my way back now.

This looks really good in principle, and it runs/builds just fine, so that's great! I'm going to take a bit of time to checkout the PR locally, get my development environment set up, and make sure I understand the implications of working in types. More in a day or two!

File: src/components/Universe.tsx [hunk 1]
```diff
@@ -23,18 +47,21 @@ function drawScatter(scatterRef, tooltipRef, setTooltipState, setTooltipData, po
     height = 4000
 
   let isMobile = winWidth < 768,
-    xx = universePosition ? universePosition.x : isMobile ? -250 : -300,
-    yy = universePosition ? universePosition.y : isMobile ? 200 : -75,
-    scale = universePosition ? universePosition.k : isMobile ? 0.2 : 0.3
+    xx = universePosition.x ?? isMobile ? -250 : -300,
```
Comment (joelhans) [inline src/components/Universe.tsx:50]: I can't believe I never knew `??` was a thing! :grimacing:

File: src/components/Universe.tsx [hunk 2]
```diff
@@ -58,71 +85,62 @@ function drawScatter(scatterRef, tooltipRef, setTooltipState, setTooltipData, po
...
-      } else {
-        return false
-      }
-    })
-    .reduce((a, { asterism, id, declination, ascension, linkedTo }) => {
-      // Loop through all linkedTo targets to support multiple links.
-      linkedTo.map((link) => {
-        a.push({
+  const links = posts
+    .filter((post) => !!post.linkedTo)
```
Comment (joelhans) [inline src/components/Universe.tsx:117]: Love how much simpler this is.

**Comment by akattow** [comment]
No worries!! I figured you were super busy with launch & opening up subs & the holidays, seems like a LOT of work! No rush at all, this was a fun break for me :)

**Comment by joelhans** [comment]
Hey @akattow! I think I'm ready to merge this—I think the best way for me to learn TypeScript is to put it into action—but it looks like a recent commit of mine created a merge conflict. If you can get that fixed without too much trouble I'll go ahead and approve/merge!

**Comment by akattow** [comment]
hey @joelhans! sorry for the delay, but the merge conflicts are resolved

**Review by joelhans — APPROVED** [review]

**Pull request merged**